include/memory_allocator.h

```diff
@@ -0,0 +1,61 @@
+#ifndef __MEMORY_ALLOCATOR_H__
+#define __MEMORY_ALLOCATOR_H__
+
+#define MAGIC 0xC0DECAFE
+#define ALIGNMENT 8
+
+TAILQ_HEAD(mb_list, mem_block);
+TAILQ_HEAD(mp_arena, mem_arena);
+
+typedef struct mem_block {
```

Comment:
I moved all the struct definitions to `memory_allocator.c`. Is that what you meant, or I did too much?